### ISSUE TITLE
2.75682452.chunk.js:2 MUI: GridErrorHandler - An unexpected error occurred. Error: MUI: useGridRootProps should only be used inside the DataGrid, DataGridPro or DataGridPremium component..  Error: MUI: useGridRootProps should only be used inside the DataGrid, DataGridPro or DataGridPremium component.[question] 

### ISSUE THREAD

**Issue by arifannosh** [mui/mui-x#6951]
### Order ID 💳

54600

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

It works fine on my local system when a deployed following error is occur  
occurring

### Your environment 🌎

<details>
   "@material-ui/core": "^4.11.4",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/lab": "^4.0.0-alpha.58",
    "@mui/icons-material": "^5.6.2",
    "@mui/material": "^5.4.4",
    "@mui/x-data-grid-premium": "^5.17.12",
npm 6.14.15
node v14.17.6

```
  chrome

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

Could you provide a working reproduction, this [Codesandbox template](https://codesandbox.io/s/datagrid-v5-quick-start-54iz1) might be a good starting point.

**Comment by arifannosh** [comment]
@flaviendelangle plz check I just update [Codesandbox template](https://codesandbox.io/s/data-grid-community-forked-iqqbq2) with my code

**Comment by arifannosh** [comment]
hello Mui/Mui-X
I just add my code to the above code sandbox.
please let me know if you want you anything else .


*Regards, *
*Arif Hussain,*
*Software Engineer*

**Comment by arifannosh** [comment]
hi Mui/Mui-X
can you please arrange a demo call?
i will show error on screen share .

*Regards, *
*Arif Hussain,*
*Software Engineer*

**Comment by arifannosh** [comment]
please have look [Example](https://codesandbox.io/s/data-grid-community-forked-jq9du5?file=/src/App.tsx:0-6413)

**Comment by arifannosh** [comment]
<GridToolbarContainer>
        {/* <GridToolbarColumnsButton
          startIcon={<ViewColumnIcon />}
          nonce={undefined}
          onResize={undefined}
          onResizeCapture={undefined}
        />
        <GridToolbarFilterButton
          nonce={undefined}
          onResize={undefined}
          onResizeCapture={undefined}
        />
        <GridToolbarDensitySelector
          nonce={undefined}
          onResize={undefined}
          onResizeCapture={undefined}
        /> */}
        <GridToolbarExport />
      </GridToolbarContainer>
      
      the custom toolbar creates an issue

**Comment by joserodolfofreitas** [comment]
Hi @arifannosh,

I did a build (with yarn build) using the `App.tsx` on your example, and I couldn't reproduce the problem. 
What's the process you're using to "deploy"?

**Comment by arifannosh** [comment]
yes @m4theushw it only gives an error on production, not on local or in a build

**Comment by joserodolfofreitas** [comment]
What do you mean by "production", @arifannosh? What's the process you're using to deploy your application?

**Comment by arifannosh** [comment]
@joserodolfofreitas and @flaviendelangle I just found a workaround 
this is an issue with **GridToolbarContainer** and **GridToolbar** components of x-grid-premium. I import those two components for x-grid-pro it works fine for me now
